source/server/server.cc

```diff
@@ -127,8 +127,9 @@ void InstanceImpl::drainListeners() {
 
 void InstanceImpl::failHealthcheck(bool fail) {
   // We keep liveness state in shared memory so the parent process sees the same state.
-  // TODO TODO TODO OH NO, CHILD->PARENT COMMUNICATION............ i guess this is fine, just make
-  // it an rpc
+  // TODO(fredlas) looks like this currently relies on shared memory for the child to convey
```

Comment:
This comment is not quite right. All this gauge is doing is indicating the current status of `/healthcheck/fail` and `/healthcheck/ok` admin commands, which end up failing or succeeding health checks for new HC connections. Given that the parent is told to drain, it enters effectively a perma-fail mode, and I think this stat is only going to be relevant in the primary process and should not be transferred from parent to child. Does that make more sense? I'm not sure how/if you want to modify this comment given that.

Reply:
Ohhh, ok, thanks for the explanation. Yes, that does make more sense; sounds like nothing to worry about.